### DISCUSSION TITLE
homebridge-google-nest-sdm config vaules roadblock

### DISCUSSION THREAD

**Discussion by RMan1971** [potmat/homebridge-google-nest-sdm#70] [Q&A]
I've gone through the step by step to get all of my config values, but when I get to the point where I can log in to my Nest and see all the devices and choose what I want in Steps 1 & 2, when I hit next to continue I get this error:

> Access blocked: This app’s request is invalid
> 
> You can’t sign in because this app sent an invalid request. You can try again later, or contact the developer about this issue.
> If you are a developer of this app, see [error details](https://accounts.google.com/).
> Error 400: redirect_uri_mismatch

I'm using the "replacement" URL from the instructions, so I'm not entirely sure where the error is occurring.  It's allowing me to log in, so the URL and my project ID and 0Auth Client ID seem to be correct.  Is the issue with my devices? Some code on the page itself?

Thanks in advance!

**Comment by noahbm** [comment]
Had the same problem today, I missed it at first, but after rereading the setup docs I was able get past it. Step 3 in [Setup Google Cloud Platform](https://developers.google.com/nest/device-access/get-started#set_up_google_cloud_platform) has you specify "https://www.google.com" as the URI redirect in the OAuth Client ID details. Give that a try and see if it helps you move on to the next step.

**Reply by vgoetten** [reply]
I was having the same issue and specifying the google URL as the URI redirect solved the issue.

**Comment by FrancisPoe** [comment]
I'm having the same issue. I'm using "https://www.google.com" has the redirect URL as instructed in step 3. However, I still get the "Error 400: redirect_uri_mismatch". Any advice?

**Comment by TassiRob** [comment]
Ran into this same issue today. Did any of you work out a fix to get past the Error 400
![error message](https://github.com/potmat/homebridge-google-nest-sdm/assets/72896259/d7cb9146-b03c-4fb4-8957-96390c8c796d)

**Reply by TassiRob** [reply]
My Bad. Did not select "web server" when setting up the web platform. 
Something to  check if others have this issue.

**Comment by TassiRob** [comment]
Select Web Server when it asks "Where are you calling from?".
